docs/geocode.rst

```diff
@@ -0,0 +1,142 @@
+Geocode
+=======
+
+The ``Geocode`` class provides geocoding using CARTO LDS.
```

Comment:
We should spell out `LDS`: `Location Data Services (LDS)`. We could also link to LDS pages that we maintain so the user has more information.